tools/path_ops/dart/lib/path_ops.dart

```diff
@@ -0,0 +1,390 @@
+          assert(false);
+          break;
+        case PathVerb._conicTo:
+          assert(false);
+          break;
+        case PathVerb.cubicTo:
+          proxy.cubicTo(
+            points[index++],
+            points[index++],
+            points[index++],
+            points[index++],
+            points[index++],
+            points[index++],
+          );
+          break;
+        case PathVerb.close:
+          proxy.close();
+          break;
+      }
+    }
+    assert(index == points.length);
+  }
+
+  /// The list of path verbs in this path.
+  ///
+  /// This may not match the verbs supplied by calls to [moveTo], [lineTo],
+  /// [cubicTo], and [close] after [applyOp] is invoked.
+  ///
+  /// This list determines the meaning of the [points] array.
+  Iterable<PathVerb> get verbs {
+    _updatePathData();
+    final int count = _pathData!.ref.verb_count;
+    return List<PathVerb>.generate(count, (int index) {
+      return PathVerb.values[_pathData!.ref.verbs.elementAt(index).value];
+    }, growable: false);
+  }
+
+  /// The list of points to use with [verbs].
+  ///
+  /// Each verb uses a specific number of points, specified by the
+  /// [pointsPerVerb] map.
+  Float32List get points {
+    _updatePathData();
+    return _pathData!.ref.points.asTypedList(_pathData!.ref.point_count);
+  }
+
+  void _updatePathData() {
+    assert(_path != null);
+    _pathData ??= _dataFn(_path!);
+  }
+
+  void _resetPathData() {
+    if (_pathData != null) {
+      _destroyDataFn(_pathData!);
+    }
+    _pathData = null;
+  }
+
+  @override
+  void moveTo(double x, double y) {
+    assert(_path != null);
+    _resetPathData();
+    _moveToFn(_path!, x, y);
+  }
+
+  @override
+  void lineTo(double x, double y) {
+    assert(_path != null);
+    _resetPathData();
+    _lineToFn(_path!, x, y);
+  }
+
+  @override
+  void cubicTo(
+    double x1,
+    double y1,
+    double x2,
+    double y2,
+    double x3,
+    double y3,
+  ) {
+    assert(_path != null);
+    _resetPathData();
+    _cubicToFn(_path!, x1, y1, x2, y2, x3, y3);
+  }
+
+  @override
+  void close() {
+    assert(_path != null);
+    _resetPathData();
+    _closeFn(_path!, true);
+  }
+
+  @override
+  void reset() {
+    assert(_path != null);
+    _resetPathData();
+    _resetFn(_path!);
+  }
+
+  /// Releases native resources.
+  ///
+  /// After calling dispose, this class must not be used again.
+  void dispose() {
+    assert(_path != null);
+    _resetPathData();
+    _destroyFn(_path!);
+    _path = null;
+  }
+
+  /// Applies the operation described by [op] to this path using [other].
+  Path applyOp(Path other, PathOp op) {
+    assert(_path != null);
+    assert(other._path != null);
+    final Path result = Path.from(this);
+    _opFn(result._path!, other._path!, op.index);
+    return result;
+  }
+}
+
+// TODO(dnfield): Figure out where to put this.
+// https://github.com/flutter/flutter/issues/99563
+final ffi.DynamicLibrary _dylib = () {
```

Comment:
My understanding is that it is more idiomatic to import (and not re-export) the stuff from here down from a library under `lib/src`.

Reply:
This is all private to the library - but I can move it.

Reply:
The problem with importing it is that people could then do source imports, which I'd like to avoid. I'm not sure which is better to handle here.

This is a basic namespacing/visibility problem in Dart though and maybe I just need to accept it?  I know in the framework we don't trust people to not do source imports.